aiohttp/client.py

```diff
@@ -987,58 +1017,73 @@ def _prepare_headers(self, headers: Optional[LooseHeaders]) -> "CIMultiDict[str]
         return result
 
     def get(
-        self, url: StrOrURL, *, allow_redirects: bool = True, **kwargs: Any
+        self,
+        url: StrOrURL,
+        **kwargs: Unpack[_RequestOptions],
     ) -> "_RequestContextManager":
         """Perform HTTP GET request."""
+        if "allow_redirects" not in kwargs:
+            kwargs["allow_redirects"] = True
+
         return _RequestContextManager(
-            self._request(hdrs.METH_GET, url, allow_redirects=allow_redirects, **kwargs)
+            self._request(
+                hdrs.METH_GET,
+                url,
+                **kwargs,
+            )
         )
 
     def options(
-        self, url: StrOrURL, *, allow_redirects: bool = True, **kwargs: Any
+        self,
+        url: StrOrURL,
+        **kwargs: Unpack[_RequestOptions],
     ) -> "_RequestContextManager":
         """Perform HTTP OPTIONS request."""
-        return _RequestContextManager(
-            self._request(
-                hdrs.METH_OPTIONS, url, allow_redirects=allow_redirects, **kwargs
-            )
-        )
+        if "allow_redirects" not in kwargs:
+            kwargs["allow_redirects"] = True
+
+        return _RequestContextManager(self._request(hdrs.METH_OPTIONS, url, **kwargs))
 
     def head(
-        self, url: StrOrURL, *, allow_redirects: bool = False, **kwargs: Any
+        self,
+        url: StrOrURL,
+        **kwargs: Unpack[_RequestOptions],
     ) -> "_RequestContextManager":
         """Perform HTTP HEAD request."""
-        return _RequestContextManager(
-            self._request(
-                hdrs.METH_HEAD, url, allow_redirects=allow_redirects, **kwargs
-            )
-        )
+        if "allow_redirects" not in kwargs:
+            kwargs["allow_redirects"] = False
+
+        return _RequestContextManager(self._request(hdrs.METH_HEAD, url, **kwargs))
 
     def post(
-        self, url: StrOrURL, *, data: Any = None, **kwargs: Any
+        self,
+        url: StrOrURL,
+        **kwargs: Unpack[_RequestOptions],
     ) -> "_RequestContextManager":
         """Perform HTTP POST request."""
-        return _RequestContextManager(
-            self._request(hdrs.METH_POST, url, data=data, **kwargs)
-        )
+        return _RequestContextManager(self._request(hdrs.METH_POST, url, **kwargs))
 
     def put(
-        self, url: StrOrURL, *, data: Any = None, **kwargs: Any
+        self,
+        url: StrOrURL,
+        **kwargs: Unpack[_RequestOptions],
     ) -> "_RequestContextManager":
         """Perform HTTP PUT request."""
-        return _RequestContextManager(
-            self._request(hdrs.METH_PUT, url, data=data, **kwargs)
-        )
+        return _RequestContextManager(self._request(hdrs.METH_PUT, url, **kwargs))
 
     def patch(
-        self, url: StrOrURL, *, data: Any = None, **kwargs: Any
```

Comment:
`None` is already the default, and it conflicts with the typed dict now.

Reply:
Not sure what this means? The type is `Any` which doesn't conflict with anything.

Reply:
Listing it out as a key-word argument, if is additionally present in the unpacked type dictionary is not allowed by type-checkers.

Reply:
e.g. this scenario, you can only define it once, it's basically equivalent to listing out an argument twice.

class MyTyped(TypedDict):
   arg_1: int

def foo(arg_1: int, **kwargs: Unpack[MyTyped]) -> None: >>.